src/test/java/RouterTest.java

```diff
@@ -0,0 +1,45 @@
+import java.util.HashMap;
+import static org.junit.Assert.assertTrue;
+import org.junit.Before;
+import org.junit.Test;
+
+
+public class RouterTest {
+  private HashMap<String, Handler> routes; 
+  private Router router;
+  
+  @Before 
+  public void setup() {
+    this.routes = new HashMap<String, Handler>();
+    String rootPath = System.getProperty("user.dir");
+    String jarDirectoryPath = rootPath + "/build/libs";
+    this.routes.put("/", new RootHandler(jarDirectoryPath));
+
+    this.router = new Router(routes);
+  }
+  
+  @Test 
+  public void returnsNotFoundHandlerWhenURIIsNotFound() {
+    Request request = new Request.Builder()
+                                 .method("GET")
+                                 .uri("/path/that/does/not/exist")
+                                 .version("1.1")
+                                 .build();
+
+    Handler handler = this.router.getHandler(request.getURI());
+    assertTrue(handler instanceof NotFoundHandler);
+  }
+  
+  @Test 
+  public void returnsRootHandler() {
+    Request request = new Request.Builder()
+                                 .method("GET")
+                                 .uri("/")
+                                 .version("1.1")
+                                 .build();
+
+   Handler handler = this.router.getHandler(request.getURI());
+   assertTrue(handler instanceof RootHandler);
+  }
```

Comment:
Generally, we don't like to assert on types because it's more brittle than asserting behavior. I understand why you wrote this because there isn't much else to test here, but you could consider returning a response from the router instead of a specific handler, and then you can test that the response is what you expect instead of just testing the type.